pkg/syncproto/sync_proto.go

```diff
@@ -102,13 +103,23 @@ type SerializedUpdate struct {
 	UpdateType api.UpdateType
 }
 
-func (s SerializedUpdate) ToUpdate() api.Update {
+var ErrBadKey = errors.New("Unable to parse key.")
+
+func (s SerializedUpdate) ToUpdate() (api.Update, error) {
 	// Parse the key.
-	parsedKey := model.KeyFromDefaultPath(string(s.Key))
-	parsedValue, err := model.ParseValue(parsedKey, s.Value)
-	if err != nil {
-		log.WithField("rawValue", string(s.Value)).Error(
-			"Failed to parse value.")
+	parsedKey := model.KeyFromDefaultPath(s.Key)
+	if parsedKey == nil {
+		log.WithField("key", s.Key).Error("BUG: cannot parse key.")
+		return api.Update{}, ErrBadKey
+	}
```

Comment:
Just a thought, but is it really better to be defensive here, or would it be better to panic on a parse failure?  We really don't want to miss any further passthrough problems that might exist.

I guess that's what we're testing for - and other than in the next few days it perhaps is better to be defensive... WDYT?